docs/CMakeLists.txt

```diff
@@ -11,14 +11,12 @@ if("${DOXYGEN_VERSION}" VERSION_GREATER_EQUAL 1.9.4 AND CMAKE_CXX_COMPILER_ID ST
 # Doxygen <= 1.9.4 suffers from https://github.com/doxygen/doxygen/issues/8784
 set(DOXYGEN_CLANG_ASSISTED_PARSING YES)
 set(DOXYGEN_CLANG_DATABASE_PATH "${CMAKE_BINARY_DIR}")
-else()
-set(DOXYGEN_WARN_AS_ERROR YES)
 endif()
 set(DOXYGEN_DISABLE_INDEX YES)
 set(DOXYGEN_ENUM_VALUES_PER_LINE 1)
 set(DOXYGEN_EXAMPLE_PATH "${DDC_SOURCE_DIR}/examples")
 set(DOXYGEN_EXPAND_ONLY_PREDEF YES)
-set(DOXYGEN_EXTRACT_ALL NO)
+set(DOXYGEN_EXTRACT_ALL YES CACHE STRING "")
```

Comment:
Why `CACHE STRING` instead of `CACHE BOOL`?

Reply:
Actually i don't know how it translates to the Doxyfile if we make it a BOOL in cmake. In the documentation they only mention `set(DOXYGEN_VAR YES)`, i don't think these local variables are typed are they ?

Reply:
As far as I could tell CMake interprets YES as a BOOL set to ON:

In the [docs for if](https://cmake.org/cmake/help/latest/command/if.html#constant) they say:
> True if the constant is 1, ON, YES, TRUE, Y, or a non-zero number (including floating point numbers).

Back when it was designed it seems that YES was a valid boolean:
https://cmake.org/pipermail/cmake/2007-December/018548.html

Reply:
I agree with you for that part but what about Doxyfile ? I was thinking of the case where cmake would dump "1" in the Doxyfile which is also a valid boolean for cmake. I don't know if this would be a valid value for a Doxyfile.

Reply:
Ah ok. Makes sense :+1: